### PR TITLE
[WIP] feat(listingSearchApi): create resolvers for listing search api

### PR DESCRIPTION
**Pull request by DongJoSuh**
- create entities
- create resolver
   - getListings(filter[ ])
       - filter { option: string, value: string }

### CODE REVIEW

File: src/resolvers/index.ts [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
 
 import { createPaymentMethod, getPaymentMethods } from './paymentMethod'
 
 export const resolvers: Resolvers = {
   Query: {
     paymentMethods: () => {
       return getPaymentMethods()
+    },
+    //@ts-ignore
+    listingsFilterByProductType: async (_, __, { filter }) => {
```
Comment (codedavinci) [inline src/resolvers/index.ts]: These filters will have to be all in one. The name of the resolver will be only `listings`

File: src/resolvers/index.ts [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
+import { getListingsFilterByProductType, getListingsFilterByTitle } from './listing'
 
 import { createPaymentMethod, getPaymentMethods } from './paymentMethod'
 
 export const resolvers: Resolvers = {
   Query: {
     paymentMethods: () => {
       return getPaymentMethods()
+    },
+    listingsFilterByProductType: async (_, __, {}) => {
```
Comment (codedavinci) [inline src/resolvers/index.ts:11]: It's meant to be one resolver for this: `listing`

File: src/resolvers/listing.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      productType: filter
+    }
+  }
+
+  const finalMatches = await ListingMongo.aggregate([_listings])
+
+  return finalMatches
+}
+
+export const getListingsFilterByTitle = async (filter: string) => {
```
Comment (codedavinci) [inline src/resolvers/listing.ts:21]: as I said yesterday these methods can be bound to the model and the filters must be passed as arguments.

File: src/schemas/entities.ts [hunk 4]
```diff
@@ -8,6 +8,8 @@ export default gql`
 
   type Query {
     paymentMethods: [PaymentMethod]!
+    listingsFilterByProductType(filter: String): [Listing]!
```
Comment (codedavinci) [inline src/schemas/entities.ts:11]: `listings(filter: filterInput): [Listing]`

**Review by codedavinci — CHANGES_REQUESTED** [review]

File: src/resolvers/index.ts [hunk 5]
```diff
@@ -1,11 +1,19 @@
...
     paymentMethods: () => {
       return getPaymentMethods()
+    },
+    //@ts-ignore
+    listings: async (_, __, { FilterInput }) => {
+      FilterInput = {
+        productType: 'TOUR',
+        title: 'Vancouver'
+      }
+      return await getListings(FilterInput)
```
Comment (codedavinci) [inline src/resolvers/index.ts:16]: `await Listing.getListings(FilterInput)`

File: src/schemas/entities.ts [hunk 6]
```diff
@@ -21,4 +22,75 @@ export default gql`
...
     updatedAt: String
   }
+
+  input FilterInput {
+    productType: String
+    title: String
+    subtitle: String
+    overview: String
+    category: [String]
+    trait: [String]
```
Comment (codedavinci) [inline src/schemas/entities.ts:32]: Add `keywords`

**Review by codedavinci — APPROVED** [review]

**Comment by DongJoSuh** [comment]
there is a block; i need calendar model to filter listing by date. going to pause on this ticket and work on ticket#36 - model Calendar

**Pull request closed without merging**